x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/config_panel.tsx

```diff
@@ -189,6 +189,7 @@ export function LayerPanels(
           onRemoveLayer={() => {
             const datasourcePublicAPI = props.framePublicAPI.datasourceLayers?.[layerId];
             const datasourceId = datasourcePublicAPI?.datasourceId;
+            const layerDatasource = datasourceMap[datasourceId];
             const layerDatasourceState = datasourceStates?.[datasourceId]
               ?.state as IndexPatternPrivateState;
```

Comment:
could you please remove casting to `as IndexPatternPrivateState;` It looks like we're trying to mask the case with `undefined`